File: lyedu-admin/src/api/exam.ts
import request from '@/utils/request'

export interface Exam {
  id: number
  title: string
  paperId: number
  startTime?: string
  endTime?: string
  durationMinutes?: number
  passScore?: number
  visibility: number
  status: number
  departmentIds?: number[]
  createTime?: string
  updateTime?: string
}

export interface ExamRecord {
  id: number
  examId: number
  userId: number
  paperId: number
  score?: number
  passed?: number
  answers?: string
  submitTime?: string
  createTime?: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  current: number
  size: number
  pages: number
}

export const getExamPage = (params: { page: number; size: number; keyword?: string }) => {
  return request.get<PageResult<Exam>>('/exam/page', { params })
}

export const getExamById = (id: number) => {
  return request.get<Exam>(`/exam/${id}`)
}

export const getExamByIdAdmin = (id: number) => {
  return request.get<Exam>(`/exam/admin/${id}`)
}

export const createExam = (data: Partial<Exam> & { paperId: number }) => {
  return request.post<number>('/exam', data)
}

export const updateExam = (id: number, data: Partial<Exam> & { paperId: number }) => {
  return request.put(`/exam/${id}`, data)
}

export const deleteExam = (id: number) => {
  return request.delete(`/exam/${id}`)
}

export const getExamRecords = (examId: number) => {
  return request.get<ExamRecord[]>(`/exam-record/exam/${examId}`)
}

File: lyedu-api/src/main/java/com/lyedu/entity/Exam.java
package com.lyedu.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;
import java.util.List;

/**
 * 考试任务实体
 *
 * @author LyEdu Team
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class Exam extends BaseEntity {

    private String title;
    private Long paperId;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    /** 考试时长（分钟），null 取试卷默认 */
    private Integer durationMinutes;
    /** 及格分，null 取试卷默认 */
    private Integer passScore;
    /** 可见性：1-公开，0-私有 */
    private Integer visibility;
    /** 状态：0-下架，1-上架 */
    private Integer status;
    /** 关联部门ID列表（私有时），来自 ly_exam_department */
    private List<Long> departmentIds;
}
